lib/Driver/ToolChains.cpp

```diff
@@ -951,7 +951,21 @@ toolchains::Darwin::constructInvocation(const LinkJobAction &job,
   const Driver &D = getDriver();
   const llvm::Triple &Triple = getTriple();
 
-  InvocationInfo II{"ld"};
+  // Configure the toolchain
+  // By default, use the system ld to link.
+  const char* LD = "ld";
+  if (const Arg *A = context.Args.getLastArg(options::OPT_tools_directory)) {
+
+    StringRef toolchainPath(A->getValue());
+
+    // If there is a 'ld' in the toolchain folder, use that instead
+    auto toolchainLD = llvm::sys::findProgramByName("ld", {toolchainPath});
+    if (toolchainLD) {
```

Comment:
Could collapse this into a if initialization:

    if (auto ld = llvm::sys::findProgramByName("ld", {toolchainPath}))
      LD = context.Args.MakeArgString(ld.get());